### PR TITLE
[Windows] Move to Visual Studio 2019

### PR DESCRIPTION
**Pull request by loic-sharma**
Use Visual Studio 2019 to build the engine. This code was modified from Chromium: https://source.chromium.org/chromium/chromium/src/+/main:build/vs_toolchain.py;l=22-48

This is the draft pull request to update the engine: https://github.com/flutter/engine/pull/36538

This is a requirement to roll ANGLE as it is affected by an STL bug if built using Visual Studio 2017: [angleproject#7693](https://bugs.chromium.org/p/angleproject/issues/detail?id=7693)

Part of: https://github.com/flutter/flutter/issues/110948

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by cbracken — APPROVED** [review]
lgtm. 

Looks like `GetVisualStudioVersion` is still used elsewhere for setting environment variables and by `DetectVisualStudioPath`, which is only used to set environment variables, some of which are used later.

Given the `GYP_` prefix, looks like they were never updated to `GN_` during the `gyp` to `gn` migration. I wonder what, if anything, other than this script depends on these. I wouldn't try to fix that in this patch and I won't mention which person claimed an apparently preemptive victory after removing the last vestiges of gyp a couple years ago.

File: build/vs_toolchain.py [hunk 1]
```diff
@@ -36,8 +36,8 @@
 
 # VS versions are listed in descending order of priority (highest first).
 MSVS_VERSIONS = collections.OrderedDict([
-  ('2017', '15.0'),
   ('2019', '16.0'),
+  ('2017', '15.0'),
```
Comment (a-wallen) [inline build/vs_toolchain.py:40]: Why was this line (2017) swapped with (2019)

**Review by stuartmorgan — APPROVED** [review]
LGTM. Looked like the mystery crash that prevented me from doing this a long time ago is gone 🎉

File: build/vs_toolchain.py (hunk 1 above)
Comment (loic-sharma) [reply build/vs_toolchain.py:40]: Yup that's right, these are ordered by priority. There's a comment above this, let me know if you have suggestions to improve clarity:

https://github.com/flutter/buildroot/blob/master/build/vs_toolchain.py#L37

**Pull request merged**